src/openforms/prefill/__init__.py

```diff
@@ -56,25 +56,31 @@ def _fetch_prefill_values(
     )
 
     @elasticapm.capture_span(span_type="app.prefill")
-    def invoke_plugin(item: Tuple[str, List[str]]) -> Tuple[str, Dict[str, Any]]:
-        plugin_id, fields = item
-        plugin = register[plugin_id]
+    def invoke_plugin(
+        item: Tuple[str, dict[str, list[str]]]
+    ) -> Tuple[str, Dict[str, Any]]:
+        plugin_id, field_groups = item
 
+        plugin = register[plugin_id]
         if not plugin.is_enabled:
             raise PluginNotEnabled()
 
-        try:
-            values = plugin.get_prefill_values(submission, fields)
-        except Exception as e:
-            logger.exception(f"exception in prefill plugin '{plugin_id}'")
-            logevent.prefill_retrieve_failure(submission, plugin, e)
-            return plugin_id, {}
-        else:
-            if values:
-                logevent.prefill_retrieve_success(submission, plugin, fields)
+        retrieved_results = {}
+        for identifier_role, fields in field_groups.items():
```

Comment:
this for-loop now runs inside the thread pool executor, so this means that if you're fetching for two different BSNs with the same plugin that the requests happen sequentially, while we should be able to do these requests in parallel.

Essentially, the processing of the `field_groups` should happen outside of this function and be fed to the `with parallel()` context manager so that we minimize the amount of sequential HTTP requests for better performance.

I think that way the actual `invoke_plugin` would not see many changes